### ISSUE TITLE
Firebase Core update causing error on Android 4.4 and 5

### ISSUE THREAD

**Issue by leossmith** [myflashlab/Firebase-ANE#145]
After updating to the latest core and adding this as suggest in the readme:

```xml
<service
    android:name="com.myflashlab.firebase.core.MyFirebaseInstanceIdService"
    android:exported="true">
    <intent-filter>
        <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
    </intent-filter>
</service>
````

bellow the provider:

```xml
<provider
						            android:name="com.google.firebase.provider.FirebaseInitProvider"
						            android:authorities="tech.simpleapps.pokecalculator.firebaseinitprovider"
						            android:exported="false"
						            android:initOrder="100" />
```

I get an error on Android 4.4 and 5, crashing the app on launch. The issue is:
```
FATAL EXCEPTION: pool-7-thread-1
Process: tech.simpleapps.pokecalculator, PID: 13172
java.lang.NullPointerException
	at com.myflashlab.firebase.core.MyFirebaseInstanceIdService.onTokenRefresh(MyFirebaseInstanceIdService.java:19)
	at com.google.firebase.iid.FirebaseInstanceIdService.zza(Unknown Source)
	at com.google.firebase.iid.FirebaseInstanceIdService.handleIntent(Unknown Source)
	at com.google.firebase.iid.zzc.run(Unknown Source)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:864)
```

**Comment by myflashlab** [comment]
you have not removed the original services, have you? This must be available in your manifest:

```xml
<service
                    android:name="com.google.firebase.iid.FirebaseInstanceIdService"
                    android:exported="true">
                    <intent-filter android:priority="-500">
                        <action android:name="com.google.firebase.INSTANCE_ID_EVENT"/>
                    </intent-filter>
                </service>
```

**Comment by myflashlab** [comment]
the error ```at com.google.firebase.iid.FirebaseInstanceIdService.zza(Unknown Source)``` means ```<extensionID>com.myflashlab.air.extensions.dependency.firebase.iid</extensionID>``` is not found! are you sure you have it added?

**Comment by leossmith** [comment]
Yes I have both the service and the extension on my code. The only change I did, was upgrading from FirebaseCore July to the one released a few days ago. Added the new service and tried it. The error only affects the version 4 and 5. It works perfectly on Android 7.

After going back to the previous .ane everything works perfectly on all devices.

**Comment by myflashlab** [comment]
We tested with Android 4 and 5 again. All working as expected!

**Comment by myflashlab** [comment]
this seems like the problem described here https://github.com/myflashlab/Firebase-ANE/issues/148 Although we were not able to duplicate the problem, we decided to put an extra check over the line. please contact us through http://www.myflashlabs.com/contact/ with your myflashlab's account username so we can send you a newer version of the firebasecore so you can test again and see if the error occurs again. Thanks for your cooperation.

**Comment by myflashlab** [comment]
V5.1.1 is now released and if the issue is similar to #148 it should be fixed now. please update your core ANE with the latest .air software.

**Comment by leossmith** [comment]
Everything works as should now :) Thank you 👍
